sdk/status.go

```diff
@@ -0,0 +1,152 @@
+package sdk
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"regexp"
+)
+
+// github status constant
+const (
+	StatusSuccess = "success"
+	StatusFailure = "failure"
+	StatusPending = "pending"
+)
+
+// context constant
+const (
+	FunctionContext = "%s"
+	StackContext    = "stack-deploy"
+	EmptyAuthToken  = ""
+	tokenKey        = "token"
+)
+
+const authTokenPattern = "^[A-Za-z0-9-_.]*"
+
+var validToken = regexp.MustCompile(authTokenPattern)
+
+type CommitStatus struct {
+	Status      string `json:"status"`
+	Description string `json:"description"`
+	Context     string `json:"context"`
+}
+
+// Status to post status to github-status function
+type Status struct {
+	CommitStatuses map[string]CommitStatus `json:"commit-statuses"`
+	EventInfo      Event                   `json:"event"`
+	AuthToken      string                  `json:"auth-token"`
+}
+
+// BuildStatus constructs a status object from event
+func BuildStatus(event *Event, token string) *Status {
+
+	status := Status{}
+	status.EventInfo = *event
+	status.CommitStatuses = make(map[string]CommitStatus)
+	status.AuthToken = token
+
+	return &status
+}
+
+// UnmarshalStatus unmarshal a status object from json
+func UnmarshalStatus(data []byte) (*Status, error) {
```

Comment:
Please add a commit for a unit test for `UnmarshalStatus` including a test where an error is generated due to a bad input.

Reply:
I have added a separate Test case to check the `UnmarshalStatus` Error cases, any other cases are getting covered here.

Reply:
I added a Separated test case for Error Check